dids/did.go

```diff
@@ -0,0 +1,96 @@
+package dids
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// DID provides a way to parse and handle Decentralized Identifier (DID) URIs
+// according to the W3C DID Core specification (https://www.w3.org/TR/did-core/).
+type DID struct {
+	// URI represents the complete Decentralized Identifier (DID) URI.
+	// Spec: https://www.w3.org/TR/did-core/#did-syntax
+	URI string
+
+	// Method specifies the DID method in the URI, which indicates the underlying
+	// method-specific identifier scheme (e.g., jwk, dht, key, etc.).
+	// Spec: https://www.w3.org/TR/did-core/#method-schemes
+	Method string
+
+	// ID is the method-specific identifier in the DID URI.
+	// Spec: https://www.w3.org/TR/did-core/#method-specific-id
+	ID string
+
+	// Params is a map containing optional parameters present in the DID URI.
+	// These parameters are method-specific.
+	// Spec: https://www.w3.org/TR/did-core/#did-parameters
+	Params map[string]string
+
+	// Path is an optional path component in the DID URI.
+	// Spec: https://www.w3.org/TR/did-core/#path
+	Path string
+
+	// Query is an optional query component in the DID URI, used to express a request
+	// for a specific representation or resource related to the DID.
+	// Spec: https://www.w3.org/TR/did-core/#query
+	Query string
+
+	// Fragment is an optional fragment component in the DID URI, used to reference
+	// a specific part of a DID document.
+	// Spec: https://www.w3.org/TR/did-core/#fragment
+	Fragment string
+}
+
+var (
```

Comment:
Might be nice to link to the grammar for this, assuming there is one.

Reply:
ah yeah! good call. done: https://github.com/TBD54566975/web5-go/commit/51dbe08f2da7b3d53b1fdd595aa1e2aa765a1c15